### PR TITLE
feat: init elements

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

introduce the `@logto/elements` package

### Background

we are starting the account center and direct sign-in elements projects, which requires us to deliver a set of ui elements for our developers. per our strategy, these elements should cover the major frameworks, or be framework-agnostic.

so Web Components is definitely a choice. besides, we can use these elements in the Logto Cloud's account center to always be the first user of these elements (i.e. eat your own dog food).

### Tech choices and rationale

- Web Components: [lit](https://github.com/lit/lit/). lit uses decorators but it's the most popular existing Web Components library. stencil is also cool but is has a [memory leak issue](https://github.com/ionic-team/stencil/issues/3607) since 2022. apparently the company has no enough resource on this project.
- bundling: [tsup](https://github.com/egoist/tsup). near-zero config, very neat and simple. nothing to complain so far.

### About this pull

init the basic setup of `@logto/elements`, and introduce this package to `@logto/console`. since this pull is already large, several things to-do in separate pulls:

- show i18n result in components (now they are just keys) #6250
- replace current console account page with elements
- figure out online documentation for elements
- consider a better approach for react component exports
- try to resolve the duplicate component registration issue in dev (hot reload)
- add tests

## Testing

image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/elements/package.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{
+  "name": "@logto/elements",
+  "version": "0.0.0",
+  "description": "Logto user interface elements.",
+  "author": "Silverhand Inc. <contact@silverhand.io>",
+  "homepage": "https://github.com/logto-io/logto#readme",
+  "license": "MPL-2.0",
+  "type": "module",
+  "private": true,
```
Comment (gao-sun) [inline packages/elements/package.json:9]: private until we release the first version

File: packages/elements/src/react.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { createComponent } from '@lit/react';
+
+import { LogtoThemeProvider, LogtoCard, LogtoFormCard } from './index.js';
+
+export const createReactComponents = (react: Parameters<typeof createComponent>[0]['react']) => {
```
Comment (gao-sun) [inline packages/elements/src/react.ts:5]: https://lit.dev/docs/frameworks/react/

File: packages/elements/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Logto elements
+
+🚧 Work in progress
```
Comment (charIeszhao) [inline packages/elements/README.md:3]: 😅 nit:
```suggestion
🚧 Working in progress
```

**Review by charIeszhao — APPROVED** [review]
LGTM

File: packages/elements/README.md (hunk 3 above)
Comment (gao-sun) [reply packages/elements/README.md:3]: https://www.merriam-webster.com/dictionary/work%20in%20progress can be a noun

**Pull request merged**